event-http/pom.xml

```diff
@@ -90,6 +90,12 @@
             <groupId>org.eclipse.jetty</groupId>
             <artifactId>jetty-server</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.eclipse.jetty.toolchain</groupId>
+                    <artifactId>jetty-servlet-api</artifactId>
```

Comment:
This artifact contains the servlet schemas, the official artifact does not.

Reply:
Is this a problem for embedded use with Jersey? I was trying to minimize the changes needed for downstream projects. We'll update to Jetty 11 with the new Jakarta namespace after updating to Jetty 10.

Reply:
Jersey can use our artifact without issue.

But we cannot use the standard artifact (`jakarta.servlet:jakarta.servlet.api`), without lots of XML loading/parsing/validation issues on the server level. (it's a valid artifact for web applications, not servers)
This was fixed in Servlet 6.0 (ee10, Jetty 12) the official artifact now has the schemas (and a valid JPMS module definition too).

Reply:
Also, the use of the standard artifact can (and does) result in your XML library hitting various external servers to obtain the XML dtds/xsds when the webapps are deployed. (even on embedded use).
This is because the XML schemas are not present in that jar and will result in the XML library needing to go fetch them from remote servers (if the public ids are uris, which not all references are).

Reply:
Is it a problem if we don't use webapps? I assume this is a new thing in Jetty 10+?

Reply:
If you use Jersey, you are technically using the Servlet API.

The "webapp" in this sense is merely the use of the Servlet API.
While you don't have to worry about all of the machinery of a war file deployment using Jetty embedded, you are not immune from the effects of the Servlet API and running a webapp.

This is the list of schemas we are talking about ...

```
javax/servlet/resources/
javax/servlet/resources/datatypes.dtd
javax/servlet/resources/j2ee_1_4.xsd
javax/servlet/resources/j2ee_web_services_1_1.xsd
javax/servlet/resources/j2ee_web_services_client_1_1.xsd
javax/servlet/resources/javaee_5.xsd
javax/servlet/resources/javaee_6.xsd
javax/servlet/resources/javaee_7.xsd
javax/servlet/resources/javaee_8.xsd
javax/servlet/resources/javaee_web_services_1_2.xsd
javax/servlet/resources/javaee_web_services_1_3.xsd
javax/servlet/resources/javaee_web_services_1_4.xsd
javax/servlet/resources/javaee_web_services_client_1_2.xsd
javax/servlet/resources/javaee_web_services_client_1_3.xsd
javax/servlet/resources/javaee_web_services_client_1_4.xsd
javax/servlet/resources/permissions_7.xsd
javax/servlet/resources/web-app_2_2.dtd
javax/servlet/resources/web-app_2_3.dtd
javax/servlet/resources/web-app_2_4.xsd
javax/servlet/resources/web-app_2_5.xsd
javax/servlet/resources/web-app_3_0.xsd
javax/servlet/resources/web-app_3_1.xsd
javax/servlet/resources/web-app_4_0.xsd
javax/servlet/resources/web-common_3_0.xsd
javax/servlet/resources/web-common_3_1.xsd
javax/servlet/resources/web-common_4_0.xsd
javax/servlet/resources/web-fragment_3_0.xsd
javax/servlet/resources/web-fragment_3_1.xsd
javax/servlet/resources/web-fragment_4_0.xsd
javax/servlet/resources/XMLSchema.dtd
javax/servlet/resources/xml.xsd
```

This contains not only war deployment schemas, but also schemas for standard XML stuff (`xml.xsd`, `XMLSchema.dtd`, `datatypes.dtd`) and things related to web services (which has nothing to do with war deployment)

Reply:
Are you saying that Jetty 10 will "result in your XML library hitting various external servers"? Is this a new behavior that is not present in Jetty 9?

I can run `io.airlift.sample.TestServer` which starts Jetty with Jersey and exercises the various JAX-RS endpoints. It works while offline, so I'm wondering how this is a problem in practice.

Do we absolutely need to update downstream projects to the new servlet API JAR when we update to Jetty 10, or is that something that can wait?

Reply:
If you are building a server, the server dependencies should be evaluated for their version is your new "minimum version" to use.

In other words ...

Scenario 1: 
Jetty 10.0.13, uses jetty-servlet-api 4.0.6.
Jersey uses Jakarta Servlet API 4.0.2.

You MUST use jetty-servlet-api 4.0.6.

Scenario 2:
Jetty 10.0.14 uses jetty-servlet-api 4.0.6
Jersey uses Jakarta Servlet API 4.0.8

Then you are free to use 4.0.6 and have no demand to upgrade to 4.0.8.

Another thing that will make you upgrade is bugs in the API (yep, this happens).
Example: Jakarta Servlet API 4.0.2 has [a bad ServletOutputStream bug](https://github.com/jakartaee/servlet/issues/228) which was fixed in Jakarta Servlet API 4.0.3.

Reply:
Thanks for explaining. I'll update everything: https://github.com/airlift/airbase/pull/334